extension/src/test/util/index.ts

```diff
@@ -21,3 +28,80 @@ export const buildMockMemento = (
       void Promise.resolve()
     }
   } as unknown as Memento)
+
+export const generateTestExpData = (data: Partial<ExpData> = {}): ExpData => ({
+  deps: null,
+  meta: { has_checkpoints: false },
+  metrics: null,
+  outs: null,
+  params: null,
+  rev: EXPERIMENT_WORKSPACE_ID,
+  timestamp: null,
+  ...data
+})
+
+export const generateTestExpState = (
+  rev: string,
+  data: Partial<ExpData> = {},
+  name?: string
+): ExpState => ({
+  data: generateTestExpData({ ...data, rev }),
+  name,
+  rev
+})
+
+export const generateWorkspaceOnlyExpShowOutput = (
+  data: Partial<ExpData> = {}
+): ExpShowOutput => [generateTestExpState(EXPERIMENT_WORKSPACE_ID, data)]
+
+const isTestExpWithName = (
+  testExp:
+    | { rev: string; name: string; data: Partial<ExpData> }
+    | Partial<ExpData>
+): testExp is { rev: string; name: string; data: Partial<ExpData> } =>
+  !!(testExp as { name: string }).name
+
+export const generateCommitWithExperiments = (
```

Comment:
[F] This is already hard to grok and understand but I actually need to make it worse.